build.gradle

```diff
@@ -671,7 +671,8 @@ subprojects {
       "-Xlint:private-shadow",
       "-Xlint:stars-align",
       "-Xlint:type-parameter-shadow",
-      "-Xlint:unused"
+      "-Xlint:unused",
```

Comment:
Pardon me, why we need those changes?

Reply:
I catch two main errors are class is depreate (1. ) and @nowarn build error  in previous build in Jenkins, but these classes or method are used by other classes. I consider that I should not migrate these methods or classes in this PR, thus I think I should closed these check when building scala version 2.3.
1.  ([Error] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR-15958/core/src/main/scala/kafka/controller/KafkaController.scala:1202:18: method setOrCreatePartitionReassignment in class KafkaZkClient is deprecated)

Reply:
The issue is caused by the magic `NotNothing`. As we are in modern scala, could you try to remove it? I give a try on my local, and both compile and test work well.

Reply:
Ok, I will remove the `"-Wconf:cat=unused-nowarn:s"`, and try to complie

Reply:
```
[2024-05-16T08:37:59.531Z] [Error] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR-15958/core/src/main/scala/kafka/network/RequestChannel.scala:174:69: @nowarn annotation does not suppress any warnings

[2024-05-16T08:37:59.531Z] one error found
```

This error happened on Jenkins.

Reply:
@m1a2st You have to remove the usage of `NotNothing`